### DISCUSSION TITLE
Gitea - use git username in terraform

### DISCUSSION THREAD

**Discussion by Ed1ks** [coder/coder#12168] [General]
Hello,

right now I am using this command for cloning from git repo:

`find "/home/${local.username}/app" -maxdepth 0 -empty -exec git clone "https://${local.username}:${data.coder_external_auth.primary-gitea.access_token}@git.domain.com/name/project.git" "/home/${local.username}/app" \;`

but this demands, that the git username equals the coder username (${local.username}:).
Is there any variable for accessing the username of the gitea user?

**Comment by matifali** [comment]
We currently only fetch and expose an OAuth `access_token` for the user, and there is no username info.

I wonder if Gitea supports being used as an OIDC provider to allow your users to sign in to Coder using Gitea.

Here are some related docs if you want to experiment.
1. Coder: https://coder.com/docs/v2/latest/admin/auth#openid-connect
2. Gitea: https://docs.gitea.com/development/oauth2-provider

**Reply by Ed1ks** [reply]
Thank you for reply.
Its possible to add gitea as provider and I've it already configurated. I am logged in Coder in Gitea. So I can use the bearer Token to authenticate git in git-server. But I would like to use gitea username for pushing commits instead of coder username.

**Reply by matifali** [reply]
What I am suggesting will make your username same as gitea username.

**Comment by matifali** [comment]
As discussed in #12704, if Gitea can add a data resource similar to GitLab's [`current_user`](https://registry.terraform.io/providers/gitlabhq/gitlab/latest/docs/data-sources/current_user), you can access the user details within terraform given only the token. 

You can try using [this](https://registry.terraform.io/providers/go-gitea/gitea/latest/docs/data-sources/user) to see if it can somehow give you the `username`. It is not well documented as of the time of writing.
